calyx-frontend/src/attributes.rs

```diff
@@ -13,8 +13,10 @@ struct HeapAttrInfo {
 
 /// Attributes associated with a specific IR structure.
 #[derive(Default, Debug, Clone)]
+#[cfg_attr(feature = "serialize", derive(serde::Serialize))]
 pub struct Attributes {
     /// Inlined attributes
+    #[cfg_attr(feature = "serialize", serde(skip))]
     inl: InlineAttributes,
```

Comment:
Why are these attributes being skipped?

Reply:
We weren't sure that `InlineAttributes` would be needed for interpreting Calyx programs since it seemed to be an internal structure, but if this is meant to be a generalized S-expression backend and it's possible folks would need it we should not skip it. If that's the case let me know and I'll un-skip it!

Reply:
`InlineAttributes` contains all the small attributes like `Reset` and `Clk` in it so you want to use that information then it needs to be serialized. I would recommend an all or nothing approach: either attributes are completely dropped in outputs or they are all available. Something in the middle like this would be confusing for people since it'll skip but not all attributes.